$OpenBSD: patch-source_tools_mudraw_c,v 1.3 2016/12/12 10:54:39 sthen Exp $

add pledge(2) to "mutool draw":
  - wpath cpath : only if output is specified

--- source/tools/mudraw.c.orig	Mon Nov 28 13:34:04 2016
+++ source/tools/mudraw.c	Mon Dec 12 09:39:02 2016
@@ -1497,6 +1497,23 @@ int mudraw_main(int argc, char **argv)
 	if (fz_optind == argc)
 		usage();
 
+	if (output && output[0] != '-' && *output != 0)
+	{
+		if (pledge("stdio rpath wpath cpath", NULL) == -1)
+		{
+			fprintf(stderr, "pledge: %s\n", strerror(errno));
+			exit(1);
+		}
+	}
+	else
+	{
+		if (pledge("stdio rpath", NULL) == -1)
+		{
+			fprintf(stderr, "pledge: %s\n", strerror(errno));
+			exit(1);
+		}
+	}
+
 	if (num_workers > 0)
 	{
 		if (uselist == 0)
